File: cpp/39d_1348A_PhoenixAnsBalance.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, a, b;
    std::vector<int> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        a = 0, b = 0;
        for(int j = 1; j < (n/2); j++){
            a += pow(2, j);
            b += pow(2,(n-j));
        }
        a += pow(2,n);
        b += pow(2, n/2);
        ans.push_back(abs(a-b));
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/39d_1348A_PhoenixAnsBalance.py
from math import pow as pow

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        a, b = 0, 0
        for i in range(1, n // 2):
            a += pow(2,i)
            b += pow(2, n - i)
        a += pow(2, n)
        b += pow(2, n//2)
        ans.append(int(abs(a - b)))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()